lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2518,7 +2518,21 @@ CompilerType TypeSystemSwiftTypeRef::GetErrorType() {
 
 CompilerType
 TypeSystemSwiftTypeRef::GetReferentType(opaque_compiler_type_t type) {
-  return m_swift_ast_context->GetReferentType(ReconstructType(type));
+  auto impl = [&]() -> CompilerType {
+    using namespace swift::Demangle;
+    Demangler dem;
+    NodePointer node = GetDemangledType(dem, AsMangledName(type));
+    if (!node ||
+        (node->getKind() != Node::Kind::Unowned &&
+         node->getKind() != Node::Kind::Unmanaged) ||
```

Comment:
should this handle `Weak` too?

Reply:
oh, there's a separate function for it?

Reply:
That surprised me too and frankly I think this API is questionable, but the SwiftASTContext implementation specifically excludes weak references!

Reply:
The other function in SwiftLanguageRuntime also unravels the Optional type that is expected in a weak reference.